tests/model_registry/rest_api/test_model_registry_secure_db.py

```diff
@@ -0,0 +1,133 @@
+import pytest
+import requests
+from typing import Self
+import copy
+from pytest_testconfig import config as py_config
+from tests.model_registry.rest_api.utils import register_model_rest_api, validate_resource_attributes
+from tests.model_registry.rest_api.constants import MODEL_REGISTER_DATA
+from tests.model_registry.utils import generate_random_name
+from tests.model_registry.constants import CA_MOUNT_PATH
+from tests.model_registry.utils import get_mr_service_by_label, get_endpoint_from_mr_service
+from kubernetes.dynamic import DynamicClient
+from utilities.constants import DscComponents, Protocols
+from ocp_resources.model_registry_modelregistry_opendatahub_io import ModelRegistry
+
+from simple_logger.logger import get_logger
+
+
+LOGGER = get_logger(name=__name__)
+
+
+@pytest.mark.parametrize(
+    "updated_dsc_component_state_scope_class",
+    [
+        pytest.param(
+            {
+                "component_patch": {
+                    DscComponents.MODELREGISTRY: {
+                        "managementState": DscComponents.ManagementState.MANAGED,
+                        "registriesNamespace": py_config["model_registry_namespace"],
+                    },
+                },
+            },
+        ),
+    ],
+    indirect=True,
+)
+@pytest.mark.usefixtures("updated_dsc_component_state_scope_class")
+class TestModelRegistryWithSecureDB:
+    """
+    Test suite for validating Model Registry functionality with a secure MySQL database connection (SSL/TLS).
+    Includes tests for both invalid and valid CA certificate scenarios.
+    """
+
+    # Implements RHOAIENG-26150
+    @pytest.mark.parametrize(
+        "patch_mysql_deployment_with_ssl_ca,patch_invalid_ca,model_registry_mysql_config,local_ca_bundle",
+        [
+            (
+                {"ca_configmap_name": "mysql-ca-configmap", "ca_mount_path": "/etc/mysql/ssl"},
+                {"ca_configmap_name": "odh-trusted-ca-bundle", "ca_file_name": "invalid-ca.crt"},
+                {"ssl_ca": f"{CA_MOUNT_PATH}/invalid-ca.crt"},
+                {"cert_name": "invalid-ca.crt"},
+            ),
+        ],
+        indirect=True,
+    )
+    @pytest.mark.usefixtures("deploy_secure_mysql_and_mr", "patch_mysql_deployment_with_ssl_ca", "patch_invalid_ca")
+    def test_register_model_with_invalid_ca(
+        self: Self,
+        admin_client: DynamicClient,
+        model_registry_namespace: str,
+        model_registry_rest_headers: dict[str, str],
+        local_ca_bundle: str,
+        deploy_secure_mysql_and_mr: ModelRegistry,
+    ):
+        """
+        Test that model registration fails with an SSLError when the Model Registry is deployed
+        with an invalid CA certificate.
+        """
+        service = get_mr_service_by_label(
```

Comment:
Why not use model_registry_instance_service or it's equivalent fixture here?

Reply:
because it is getting the value of `MR_INSTANCE_NAME` . it should be to parametrize, but I prefer to not do in this PR that is already big

Reply:
Ok, let's handle it in a follow up PR.